great_expectations/data_context/data_context/base_data_context.py

```diff
@@ -1913,6 +1913,22 @@ def _instantiate_datasource_from_config(
             )
         return datasource
 
+    def update_datasource(
+        self,
+        datasource_name: str,
+        datasource_config: DatasourceConfig,
+    ) -> None:
+        """
+        Updates a DatasourceConfig that already exists in the store.
+
+        Args:
+            datasource_name: The name of the Datasource to update.
+            datasource_config: The config object to persist using the DatasourceStore.
+        """
```

Comment:
Do you want to add a `Raises` section here or do you not want to bake that into the API?

Reply:
That's a good question! I'm not sure what is a best practice here. I've leaned towards only adding `Raises` if an exception is raised within the top level scope of the function body.

Any preference?